lib/intacct/base.rb

```diff
@@ -54,6 +56,20 @@ def persisted?
       !!attributes["recordno"]
     end
 
+    protected
+
+    def format_date(date_object)
+      return if date_object.blank?
+
+      if date_object.is_a?(Date) || date_object.is_a?(Time)
+        date_object.strftime(DATE_FORMAT)
+      elsif date_object.is_a?(String)
+        Date.strptime(date_object, DATE_FORMAT).strftime(DATE_FORMAT)
```

Comment:
Could this cause issues if we are parsing a "time" string to a date (losing precision)? Would it be okay to just default to the `else` case and return the raw string?

Reply:
No fields that we work have a time component, so I wouldn't know which format to parse the time in.

Reply:
Ah right, I forgot we are explicitly calling `format_date` on specific fields anyway 🤦.